rules/do-you-return-detailed-error-messages/rule.md

```diff
@@ -17,23 +19,103 @@ Good error design is as important to the sucess of an API as the API design itse
 
-### ...And then include the problem details
-RFC 7807 - Problem Details for HTTP APIs (ietf.org) details the specification for returning errors from your API. The HTTP Status Codes are an excellent start - they immediately tell you *where*the problem is, but they don't tell you *what*the problem is.
+### Use ProblemDetails Format
+[RFC 7807 - Problem Details for HTTP APIs](https://www.rfc-editor.org/rfc/rfc7807.html) details the specification for returning errors from your API. 
 
-ASP.Net Core has built in support for the problem details specification. You can see more at the official documentation. Handle errors in ASP.NET Core web APIs | Microsoft Docs
+Problem Details defines a standardised way for an HTTP APIs to communicate errors to clients. It introduces a simple and consistent format for describing errors, providing developers with a clear and uniform way to understand and handle errors in HTTP APIs.
 
-### And for any API (REST, gRPC and GraphQL):
+Below is an example of an error message in Problem Details format:
+```json
+{
+  "type": "https://example.com/probs/invalid-id",
+  "title": "Invalid ID",
+  "status": 400,
+  "detail": "The provided ID has invalid characters.",
+  "instance": "/account/12%203",
+  "allowedCharacters": "^[a-zA-Z0-9]+$"
+}
+```
+
+In the above example:
+- `type` specifies a URI that uniquely identifies the type of the problem.
+- `title` provides a short, human-readable summary of the problem.
+- `status` indicates the HTTP status code for the response.
+- `detail` gives a human-readable explanation specific to the occurrence of the problem.
+- `instance` provides a URI reference that identifies the specific occurrence of the problem.
+- `allowedCharacters` is an example property specificly added to the problem.
+
+Using the above structured message, APIs can now reliably communicate with clients to provide a better error handling.
+
+
+### Use .NET Exception Handler
+
+ASP.NET Core has built in support for the problem details specification since .NET 7.
+
+### Option 1 - Use built-in ProblemDetails service
+
+```csharp
+// Program.cs
+// This adds ProblemDetails service
+// Read more on https://learn.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-8.0#problem-details
+builder.Services.AddProblemDetails();
+
+...
+
+// This instructs the API to use built-in exception handler
+app.UseExceptionHandler();
+```
+
+Using this option, the API will generate a problem details response for all HTTP client and server error responses that *don't have a body content yet*.
+
+You can also customise the `ProblemDetailsService` behaviour - read more about it in the following link [Handle errors in ASP.NET Core | Customise Problem Details](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/error-handling?view=aspnetcore-8.0#customize-problem-details).
+
+
+⚠️ **Important**
+When there is an exception, this approach will responds in ProblemDetails format **only in Production mode**!
```

Comment:
I would change this to:
The built in .NET ProblemDetails service will **only** produce ProblemDetails responses for exceptions in release configuration.

Reply:
Done - updated for more clarity